File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Convert/ConversionStage.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Convert;

namespace Db4objects.Db4o.Internal.Convert
{
	/// <exclude></exclude>
	public abstract class ConversionStage
	{
		public sealed class ClassCollectionAvailableStage : ConversionStage
		{
			public ClassCollectionAvailableStage(LocalObjectContainer file) : base(file)
			{
			}

			public override void Accept(Conversion conversion)
			{
				conversion.Convert(this);
			}
		}

		public sealed class SystemUpStage : ConversionStage
		{
			public SystemUpStage(LocalObjectContainer file) : base(file)
			{
			}

			public override void Accept(Conversion conversion)
			{
				conversion.Convert(this);
			}
		}

		private LocalObjectContainer _file;

		protected ConversionStage(LocalObjectContainer file)
		{
			_file = file;
		}

		public virtual LocalObjectContainer File()
		{
			return _file;
		}

		public virtual int ConverterVersion()
		{
			return _file.SystemData().ConverterVersion();
		}

		public abstract void Accept(Conversion conversion);
	}
}

File: db4oj/core/src/com/db4o/internal/convert/ConversionStage.java
/* Copyright (C) 2004 - 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.convert;

import com.db4o.internal.*;

/**
 * @exclude
 */
public abstract class ConversionStage {
	
	public final static class ClassCollectionAvailableStage extends ConversionStage {
		
		public ClassCollectionAvailableStage(LocalObjectContainer file) {
			super(file);
		}

		public void accept(Conversion conversion) {
			conversion.convert(this);
		}
	}

	public final static class SystemUpStage extends ConversionStage {
		public SystemUpStage(LocalObjectContainer file) {
			super(file);
		}
		public void accept(Conversion conversion) {
			conversion.convert(this);
		}
	}

	private LocalObjectContainer _file;
	
	protected ConversionStage(LocalObjectContainer file) {
		_file = file;
	}

	public LocalObjectContainer file() {
		return _file;
	}

	public int converterVersion() {
		return _file.systemData().converterVersion();
	}
	
    public abstract void accept(Conversion conversion);

}
